calcfetch

```diff
@@ -19,53 +19,49 @@ case $manager in
     apk) packages="$(apk list --installed | wc -l)";;
     *) packages="idk"
 esac
-shell="$(basename "$SHELL")"
+
+shell=${SHELL##*/}
+
 while :; do
     location="$(readlink "/proc/${ppid:-$$}/exe")"
     case $location in
-        *sh) :;;
+        *sh) ;;
         *) break;;
     esac
     ppid=$(grep -i ppid "/proc/${ppid:-$$}/status" | grep -o '[0-9]\+')
 done
 term="${location##*/}"
-# echo $term
 
 ## UPTIME DETECTION
 
-s=$(< /proc/uptime)
-s=${s/.*}
-d="$((s / 60 / 60 / 24)) days"
-h="$((s / 60 / 60 % 24)) hours"
-m="$((s / 60 % 60)) mins"
+IFS=. read -r s _ < /proc/uptime
+d="$((s / 60 / 60 / 24))"
+h="$((s / 60 / 60 % 24))"
+m="$((s / 60 % 60))"
 
-# Remove plural if < 2.
-((${d/ *} == 1)) && d=${d/s}
-((${h/ *} == 1)) && h=${h/s}
-((${m/ *} == 1)) && m=${m/s}
+# Plurals
+[ "$d" -gt 1 ] && dp=s
+[ "$h" -gt 1 ] && hp=s
+[ "$m" -gt 1 ] && mp=s
 
 # Hide empty fields.
-((${d/ *} == 0)) && unset d
-((${h/ *} == 0)) && unset h
-((${m/ *} == 0)) && unset m
-
-uptime=${d:+$d, }${h:+$h, }$m
-uptime=${uptime%', '}
-uptime=${uptime:-$s secs}
+[ "$d" = 0 ] && d=
+[ "$h" = 0 ] && h=
+[ "$m" = 0 ] && m=
```

Comment:
The second bit was a sideaffect of you going straight to variable substution.

Just set $uptime later on.